### PR TITLE
add chpter3 linear-networks of paddle

### PR DESCRIPTION
**Pull request by w5688414**

### CODE REVIEW

File: chapter_linear-networks/index.md [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+:maxdepth: 2
+
+linear-regression
+linear-regression-scratch
+linear-regression-concise
+softmax-regression
+image-classification-dataset
+softmax-regression-scratch
+softmax-regression-concise
+```
```
Comment (AnirudhDagar) [inline chapter_linear-networks/index.md:21]: I don't understand why this file needs changing?

File: chapter_linear-networks/linear-regression-concise.md [hunk 2]
```diff
@@ -1,424 +1,485 @@
...
+from torch.utils import data
+```
+
+```{.python .input}
+#@tab tensorflow
+from d2l import tensorflow as d2l
+import numpy as np
+import tensorflow as tf
+```
+
```
Comment (AnirudhDagar) [inline chapter_linear-networks/linear-regression-concise.md:40]: This part should not be in the diff. We need to add only the paddle code blocks. Not change what was already there. I don't see any change here but it still shows up in the diff for some reason.

File: chapter_linear-networks/linear-regression-concise.md [hunk 3]
```diff
@@ -1,424 +1,485 @@
...
+#@tab tensorflow
+def load_array(data_arrays, batch_size, is_train=True):  #@save
+    """构造一个TensorFlow数据迭代器"""
+    dataset = tf.data.Dataset.from_tensor_slices(data_arrays)
+    if is_train:
+        dataset = dataset.shuffle(buffer_size=1000)
+    dataset = dataset.batch(batch_size)
+    return dataset
+```
+
```
Comment (AnirudhDagar) [inline chapter_linear-networks/linear-regression-concise.md:86]: same

**Review by AnirudhDagar — COMMENTED** [review]
@tngt @w5688414 paddle.py file changes look fine in this PR. But I'm not sure why all other notebooks have huge diffs? These notebooks should only add paddle code blocks. But in the diff I can see that there are other lines too.


I commented on a couple of notebooks about the issue but there are almost all the notebooks with this problem.

**Comment by tngt** [comment]
> I can try to list all the steps from scratch again. I'm sorry for being nit-picky and apologize if the d2lbook dev cycle is not very straightforward.
> 
>     * `git clone git@github.com:d2l-ai/d2l-zh.git`
> 
>     * `cd d2l-zh`
> 
>     * `git checkout paddle`
> 
>     * Create a conda environment: `conda env create -n d2l-zh python=3.8 -y`
> 
>     * `conda activate d2l-zh`
> 
>     * `pip install sphinx` (Install this before, because d2lbook installation may fail without sphinx; see [[Bug] ModuleNotFoundError: No module named 'sphinx' d2l-book#48](https://github.com/d2l-ai/d2l-book/issues/48) )
> 
>     * Update environment to install requirements via yaml file: `conda env update -n d2l-zh -f static/build.yml`
> 
> 
> Note that we don't edit markdown files directly, we open them as notebooks and make the edits there. See below:
> 
>     * Inside d2l-zh repo with paddle branch checked out, run: `jupyter notebook --NotebookApp.contents_manager_class='notedown.NotedownContentsManager'`
> 
>     * Now add the paddle code by opening up each notebook that has code in ch3. Make sure you add the `#@save` ticks wherever required so that the function is also saved in the paddle.py file later.
> 
>     * Make sure all the notebooks are saved after clearing the outputs (Use restart and clear output)
> 
>     * Then run `d2lbook activate all {chapter}/{notebook}.md` to activate all the notebook tabs.
> 
>     * Run `d2lbook build lib`, this should only modify `d2l/paddle.py` in your case since we are only adding code for paddle and not changing anything on other frameworks
> 
>     * `git add {chapter_name} d2l/paddle.py`
> 
>     * `git commit -m "[Paddle] Add chapter {name}"`

@AnirudhDagar Thank you very much for your kind guidance and sorry for the inconvenience we have caused you.
@w5688414 Can you follow this process and try again?

**Comment by tngt** [comment]
Hi@AnirudhDagar I didn't notice a big difference, I know it looks like there are a lot of changes in green, that's because our team members helped organize the content, we just copied the files in, so it showed a big diff, but only the code is modified in fact。
Does this bring inconvenience to your review?

**Comment by AnirudhDagar** [comment]
Yes, it is slightly hard to review and understand the diffs. But let's say that's still manageable, and I'm fine reviewing the PR in the current state. The problem here is about polluting the diff and later causing difficulties that this causes in `git blame`.

So consider this example:
At some point of time in the future, we discover a bug, and sometimes to understand the fix, we need to pinpoint where it originated. In such a case, one heavily relies on `git blame` as a useful tool to understand the changes made to a each line in a file.

If we merge this PR, it will change almost everything in these files as seen in the diff right now (even though there are no actual changes) and later it would be tough to understand git blame or pinpoint an issue as I mentioned in the above scenario. I hope you understand what my concern is.

Lastly, as an unwritten rule for any open-source project, we almost never want to push changes that we do not intend to make (here we have a big diff) so that PRs are ideally smaller (sometimes it may be unavoidable).

**Comment by tngt** [comment]
> Yes, it is slightly hard to review and understand the diffs. But let's say that's still manageable, and I'm fine reviewing the PR in the current state. The problem here is about polluting the diff and later causing difficulties that this causes in `git blame`.
> 
> So consider this example: At some point of time in the future, we discover a bug, and sometimes to understand the fix, we need to pinpoint where it originated. In such a case, one heavily relies on `git blame` as a useful tool to understand the changes made to a each line in a file.
> 
> If we merge this PR, it will change almost everything in these files as seen in the diff right now (even though there are no actual changes) and later it would be tough to understand git blame or pinpoint an issue as I mentioned in the above scenario. I hope you understand what my concern is.
> 
> Lastly, as an unwritten rule for any open-source project, we almost never want to push changes that we do not intend to make (here we have a big diff) so that PRs are ideally smaller (sometimes it may be unavoidable).

Ok, I completely understand, thank you very much for your answer, we will strictly follow your steps to PR～

**Pull request closed without merging**